File: 169.py
class Solution(object):
    def majorityElement(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        count, majority = 1, nums[0]
        for item in nums[1:]:
            if majority!=item:
                if count ==0:
                    count = 1
                    majority = item
                else:
                    count -= 1
            else:
                count += 1
        return majority

File: 169.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int count = 1;
        int majority = nums[0];
        for (int i = 1; i < nums.size(); i++){
            if (majority == nums[i]){
                count++;
            }else{
                if (count == 0){
                    count = 1;
                    majority = nums[i];
                }else{
                    count--;
                }
            }
        }
        return majority;
    }
};
